### PR TITLE
[GRW-869] Add segment to sauce docs

### PR DESCRIPTION
**Pull request by ethanj-sl**
### Description
Add a "Page Viewed" event to Sauce Docs to help track user journeys.

### Motivation and Context
The docs site includes a lot of useful information and guidance for our customers (and potential customers) but we're currently unable to analyze how performant the content is, where users 'drop off', and what actions do the users go on to perform or not perform. As such, we'd like to add an event to track which pages users view.
https://saucedev.atlassian.net/browse/GRW-833

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation fix (typos, incorrect content, missing content, etc.)

### Checklist

- [x] I have read the [contributing](https://github.com/saucelabs/sauce-docs/blob/main/docs/contributing.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/theme/Segment.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { useLocation } from '@docusaurus/router';
+
+import '@site/static/analytics.js'
+
+const Segment = () => {
+  const { pathname } = useLocation();
+  
+  useEffect(() => {
+    if (window?.analytics) {
+      window.analytics.track('Page Viewed', {
```
Comment (VeraWei) [inline src/theme/Segment.js:11]: How about `window?.analytics?.track`?

File: src/theme/Segment.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        page: window.location.pathname,
+        domain: window.location.origin,
+        subject_name: 'Sauce Docs',
+        product_area: 'Docs',
+        product_sub_area: 'Sauce Docs',
+      });
+    }
+  }, [pathname]);
+
```
Comment (VeraWei) [inline src/theme/Segment.js:20]: What if we want to do other events tracking, e.g. `Object Clicked`?

File: static/analytics.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
...
-analytics.page();
-}}();
\ No newline at end of file
+!function(){var analytics=window.analytics=window.analytics||[];if(!analytics.initialize)if(analytics.invoked)window.console&&console.error&&console.error("Segment snippet included twice.");else{analytics.invoked=!0;analytics.methods=["trackSubmit","trackClick","trackLink","trackForm","pageview","identify","reset","group","track","ready","alias","debug","page","once","off","on","addSourceMiddleware","addIntegrationMiddleware","setAnonymousId","addDestinationMiddleware"];analytics.factory=function(e){return function(){var t=Array.prototype.slice.call(arguments);t.unshift(e);analytics.push(t);return analytics}};for(var e=0;e<analytics.methods.length;e++){var key=analytics.methods[e];analytics[key]=analytics.factory(key)}analytics.load=function(key,e){var t=document.createElement("script");t.type="text/javascript";t.async=!0;t.src="https://cdn.segment.com/analytics.js/v1/" + key + "/analytics.min.js";var n=document.getElementsByTagName("script")[0];n.parentNode.insertBefore(t,n);analytics._loadOptions=e};analytics._writeKey="13rd58d6s3";analytics.SNIPPET_VERSION="4.13.2";
+analytics.load("13rd58d6s3");
+analytics.identify();
```
Comment (VeraWei) [inline static/analytics.js:3]: Is this auto-generated? the string in `analytics.load` is key?

File: src/theme/Segment.js (hunk 2 above)
Comment (dpgraham) [reply src/theme/Segment.js:20]: The scope of this ticket is just to do page views, for now

Comment (ethanj-sl) [reply src/theme/Segment.js:20]: Add an `onClick` event handler and call the appropriate event

File: static/analytics.js (hunk 3 above)
Comment (ethanj-sl) [reply static/analytics.js:3]: Yeah, the string in `load()` is the key.
https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/quickstart/#step-2-add-the-segment-snippet

I swapped the `page` call for `identify` as the `page` calls are blocked and `identify` will allow us to see who is browsing the docs page (given that they logged into the sauce app prior)

**Review by VeraWei — APPROVED** [review]

**Review by dpgraham — APPROVED** [review]

**Pull request merged**